main.tf

```diff
@@ -1,5 +1,15 @@
 data "aws_caller_identity" "current" {}
 
+resource "aws_s3_bucket_notification" "bucket_notification" {
+  count  = var.notification_sns_arn != [""] && var.notification_events != [""] ? 1 : 0
```

Comment:
`notification_sns_arn` is going to be a string so `""` is fine there, but because `notification_events` is going to be a list of strings that's why I suggested you set it as `[""]`.

Reply:
This won't solve the issue as it's stating that the arn can't be null on the test. We probably need to set the default arn to something it can recognise. If there are no "[]" in the then it appears happy.

Reply:
Will look